fili-core/src/main/java/com/yahoo/bard/webservice/druid/client/impl/AsyncDruidWebServiceImplV2.java

```diff
@@ -0,0 +1,173 @@
+            return requestBuilder.execute(
+                    new AsyncCompletionHandler<Response>() {
+                        @Override
+                        public Response onCompleted(Response response) {
+                            RequestLog.restore(logCtx);
+                            RequestLog.stopTiming(timerName);
+                            if (outstanding.decrementAndGet() == 0) {
+                                RequestLog.startTiming(RESPONSE_WORKFLOW_TIMER);
+                            }
+                            String druidQueryId = response.getHeader("X-Druid-Query-Id");
+                            RequestLog.record(new DruidResponse(druidQueryId));
+
+                            javax.ws.rs.core.Response.Status status = javax.ws.rs.core.Response.Status.fromStatusCode(
+                                    response.getStatusCode()
+                            );
+                            LOG.debug(
+                                    "druid {} response code: {} {} and druid query id: {}",
+                                    getServiceConfig().getNameAndUrl(),
+                                    status.getStatusCode(),
+                                    status,
+                                    druidQueryId
+                            );
+
+                            if (status != javax.ws.rs.core.Response.Status.OK) {
+                                getHttpErrorMeter().mark();
+                                LOG.debug(
+                                        "druid {} error: {} {} {} and druid query id: {}",
+                                        getServiceConfig().getNameAndUrl(),
+                                        status.getStatusCode(),
+                                        status.getReasonPhrase(),
+                                        response.getResponseBody(),
+                                        druidQueryId
+                                );
+
+                                error.invoke(
+                                        status.getStatusCode(),
+                                        status.getReasonPhrase(),
+                                        response.getResponseBody()
+                                );
+                            } else {
+                                MappingJsonFactory jsonFactory = new MappingJsonFactory();
+                                try {
+                                    JsonNode rootNode;
+                                    ObjectNode objectNode;
+                                    try (
+                                            JsonParser jsonParser = jsonFactory.createParser(
+                                                    response.getResponseBodyAsStream()
+                                            )
+                                    ) {
+                                        rootNode = jsonParser.readValueAsTree();
+                                        objectNode = (ObjectNode) rootNode;
```

Comment:
I would try out and consider moving the response parsing into jsonNode process into its own method in AsyncDruidWebserviceImpl.java and simply override it here instead of overriding the whole sendRequest method. If we do that, we also need to move the response status check on the top of this sendRequest method into its own method and override it here also. If you think this makes sense too, try it.